### ISSUE TITLE
Remove imports requirement

### ISSUE THREAD

**Issue by kentcdodds** [weyert/mdx.macro#2]
I think you should be able to access the program node which you can use to access the body and insert the imports manually without requiring people to call the `imports()` function. Wanna give that a try?

**Comment by weyert** [comment]
It's an optional function `imports()` which you don't need to call. Do I understand you correctly that you are suggesting that I can access the file's main node and add these import statements automatically?

**Comment by kentcdodds** [comment]
Yup! Like this: https://github.com/hamlim/inline-mdx.macro/issues/9

**Comment by weyert** [comment]
Great, thank you! I will have a look at it 🤓

**Comment by weyert** [comment]
I am currently stuck that mdx suddenly generates a class instead of a functional component so the code breaks at the moment. Not sure what's triggering it returning a class 🤔 Ever seen this before, @kentcdodds?

**Comment by kentcdodds** [comment]
Nope 🙃

**Comment by weyert** [comment]
Looks like mdx changed it in v16.1.0 👎
